charts/metering-ansible-operator/templates/crds/meteringconfig.crd.yaml

```diff
@@ -156,6 +157,44 @@ spec:
                         serviceAccountKeyJSON:
                           type: string
                           minLength: 1
+                    s3Compatible:
+                      type: object
+                      required:
+                        - bucket
+                        - endpoint
+                      oneOf:
+                      - required:
+                        - accessKeyID
+                        - secretAccessKey
+                        allOf:
+                        - not:
+                            required:
+                            - secretName
+                      - required:
+                        - secretName
```

Comment:
Do we not need to describe the properties for this sub-schema, i.e.
```
- required:
  - secretName
  properties:
    createSecret:
      enum:
      - false
    ...
```

Reply:
I have a separate check in ansible so we can have it but it will be redundant

Reply:
yeah, most of the CRD schema validates things that the operator checks in the ansible role (and the go-code), but the idea here is that we validate it statically. It's much quicker to validate it here, then punt the validation to the operator/ansible, despite the checks being redundant.

Reply:
Agreed, we want as much as possible done in the CRD. Ansible logic is a hold-over, and may be removed eventually, but we'll keep it for now.